### ISSUE TITLE
60 upvalue limit

### ISSUE THREAD

**Issue by NanoBob** [yanghuan/CSharp.lua#226]
In more recent versions of the compiler we've encountered the following issue:
```
...\Slipe\Core\Lua\Compiled\Server\Source\SlipeShared\Vehicles\Handling.lua:488: function at line 420 has more than 60 upvalues
```

This has to do with the fact that all methods and properties (in our case properties) are marked as locals, and there's a limit on that. Storing all of these locals in a table would resolve the issue.

For example instead of:
```lua
local getX, getY, getY, getW -- etc

getX = function()
    return 0
end -- etc
```
Compile to:
```lua
local locals = { }
locals.getX = function()
    return 0
end -- etc
```

Would it be possible to make this change?

**Comment by yanghuan** [comment]
can you show this code?  and what is you use lua version?

**Comment by NanoBob** [comment]
An example class that results in this error can be found at:
https://github.com/mta-slipe/Slipe-Core/blob/development/Slipe/Core/Source/SlipeClient/Vehicles/VehicleModel.cs

The generated lua for that class is:
https://gist.github.com/NanoBob/7edf00f194d74f23fcf8a93b0ad18f00


But also the following class:
https://github.com/mta-slipe/Slipe-Core/blob/development/Slipe/Core/Source/SlipeShared/Vehicles/Handling.cs
Has the same issue, compiled lua for that being:
https://gist.github.com/NanoBob/1a8f663423bd7e09e66f179e64c9d07c

**Comment by NanoBob** [comment]
Your commit has fixed this for several files, however there is still one file in our project which still results in the "more than 60 upvalues" error.

The C# source for this is:
https://github.com/mta-slipe/Slipe-Core/blob/development/Slipe/Core/Source/SlipeShared/Utilities/Color.cs

The compiled Lua is:
https://gist.github.com/NanoBob/af1dabd92bdf4daf1831dfa878c05534

```
...re\Slipe\Core\Lua\Compiled\Server\Source\SlipeShared\Utilities\Color.lua:929: function at line 771 has more than 60 upvalues
```

**Comment by yanghuan** [comment]
fix 41238df915c01851bdc4368c86fb3507ebc2791b

**Comment by NanoBob** [comment]
Thanks, it works!
